### ISSUE TITLE
[stable/postgresql] Pods fail to create

### ISSUE THREAD

**Issue by engineertdog** [helm/charts#16898]
**Describe the bug**
I attempted to deploy v6.3.4 of the stable postgresql chart with HA (replication) and ended up with a deployment that could not create the master pod nor the slave pod. The error messages from each pod are listed below, along with the override values.

**Version of Helm and Kubernetes**:
Helm v2.14.2
Kubernetes v1.13

**What you expected to happen**:
I expect all master pods and slave pods to be created for a HA deployment.

**How to reproduce it** (as minimally and precisely as possible):
- helm create chart_override_name
    - Delete default templates and values.
- helm fetch stable/postgresql
    - unzip stable chart
- Move postgresql chart into custom chart's `chart` folder.
- Use the override values provided below.

**Anything else we need to know**:
Master pod error:
`Error from server (BadRequest): a container name must be specified for pod app-db-postgresql-master-0, choose one of: [app-db-postgresql metrics] or one of the init containers: [init-chmod-data]`

Slave pod error:
`/entrypoint.sh: line 15: nENk: unbound variable`

Chart override
```
postgresql:
  postgresqlUsername: postgresql
  postgresqlPassword: xyz
  postgresqlDatabase: newdb
  replication:
    enabled: true
    slaveReplicas: 3
    synchronousCommit: "on"
    numSynchronousReplicas: 1
    applicationName: app-db
  metrics:
    enabled: true
```

**Comment by juan131** [comment]
Hi @engineertdog 

There's a small bug in your **values.yaml**. The `replication.applicationName` parameter should use underscore instead of hyphen:

```diff
    synchronousCommit: "on"
    numSynchronousReplicas: 1
-   applicationName: app-db
+   applicationName: app_db
  metrics:
```

Using the right values, I was able to run PostgreSQL without issues:

```bash
$ helm repo update
Hang tight while we grab the latest from your chart repositories...
...Skip local chart repository
...Successfully got an update from the "bitnami" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete.
$ helm install stable/postgresql --name my-release -f values.yaml
...
$ kubectl logs my-release-postgresql-master-0 my-release-postgresql
...
postgresql 09:01:23.45 INFO  ==> ** Starting PostgreSQL **
2019-09-06 09:01:23.466 GMT [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2019-09-06 09:01:23.466 GMT [1] LOG:  listening on IPv6 address "::", port 5432
2019-09-06 09:01:23.469 GMT [1] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2019-09-06 09:01:23.487 GMT [212] LOG:  database system was shut down at 2019-09-06 09:01:22 GMT
2019-09-06 09:01:23.492 GMT [1] LOG:  database system is ready to accept connections
2019-09-06 09:01:28.037 GMT [230] LOG:  standby "app_db" is now a synchronous standby with priority 1
$ kubectl get pods
NAME                             READY   STATUS    RESTARTS   AGE
my-release-postgresql-master-0   2/2     Running   0          11m
my-release-postgresql-slave-0    1/1     Running   0          11m
my-release-postgresql-slave-1    1/1     Running   0          11m
my-release-postgresql-slave-2    1/1     Running   0          10m
```

**Comment by engineertdog** [comment]
@juan131

Ah, thank you for that information. I noticed the underscore in the **values.yaml** file in the chart, but I was unaware that there could not be a hyphen as it’s usually not an issue. I’ll test later when I’m home from work to verify the issue so that this can be closed.

Maybe a small note could be added to the **readme** and **values** files indicating the requirement?

**Comment by juan131** [comment]
Hi @engineertdog 

I went a step further and I found the issue why the hyphens are not accepted. I created a PR to fix the issue in the image, find it in the link below:

- https://github.com/bitnami/bitnami-docker-postgresql/pull/167

**Comment by engineertdog** [comment]
I'm able to get it up and running now.

It looks like another issue I was having was in relation to the `postgresqlPassword` that I was trying to set. With the randomly generated password, I had special characters that weren't allowed, not even encapsulated with quotes. This was what was throwing the `/entrypoint.sh: line 15: nENk: unbound variable` error.

**Comment by juan131** [comment]
Hi @engineertdog 

That's weird. We faced issues in the past with password that weren't correctly escaped but we already fixed them.

It could be related with some other configuration parameter. Could you share the **values.yaml** you're using so I can reproduce it locally?

**Comment by engineertdog** [comment]
It's the same **values.yaml** override that I listed above, with the exception that I ended up using double quotes for the password, like `"some$randomPassword!1234"`. I did confirm that it was the password field that was causing the issue, but I didn't troubleshoot which character was the problem, or if it was all of them. I removed all of the special characters at once and the chart worked.

I do not remember all of the special characters I had in the password, but I did have the following: `!$#*`
